notebooks/amazon_forest_notebook.py

```diff
@@ -190,119 +191,121 @@
-# 
-# __CSVLogger__
-# 
-# Creates a file with a log of loss and accuracy per epoch
-# 
-# __FBeta__
-# 
-# Calculates fbeta_score after each epoch during training
+# Here we define the model for finetuning
 
 # <codecell>
 
-checkpoint = ModelCheckpoint(filepath="weights.best.hdf5", monitor='val_acc', verbose=1, save_best_only=True)
+from keras.applications.vgg16 import VGG16
+from keras.models import Model
+from keras.layers import Dense, Input, Flatten, Dropout
+from keras.layers.normalization import BatchNormalization
 
-# creates a file with a log of loss and accuracy per epoch
-csv = CSVLogger(filename='training.log', append=True)
+# VGG16
 
-# Calculates fbeta_score after each epoch during training
-fbeta = Fbeta()
+img_dim = (128, 128, 3)
 
-# <markdowncell>
+input_tensor = Input(shape=img_dim)
 
-# ## Choose Hyperparameters
-# 
-# Choose your hyperparameters below for training. 
-# 
-# Note that we have created a learning rate annealing schedule with a series of learning rates as defined in the array `learn_rates` and corresponding number of epochs for each `epochs_arr`. Feel free to change these values if you like or just use the defaults.
-# 
-# If you opted for a high `img_resize` then you may want to lower the `batch_size` to fit your images matrices into the GPU memory. With an `img_resize` of `(256, 256)` (the default size) and a batch_size of `64` you should at least have a GPU with 8GB or VRAM.
+base_model = VGG16(include_top=False,
+                   weights='imagenet',
+                   input_shape=img_dim)
 
-# <codecell>
+bn = BatchNormalization()(input_tensor)
+x = base_model(bn)
+x = Flatten()(x)
+output = Dense(17, activation='sigmoid')(x)
+
+model = Model(input_tensor, output)
 
-batch_size = 64
-epochs_arr = [35, 15, 5]
-learn_rates = [0.002, 0.0002, 0.00002]
+model.summary()
 
 # <markdowncell>
 
-# ## Define and Train model
-# 
-# Here we define the model and begin training. 
+# ## Fine-tune conv layers
+# Now that we have a trained classifier, we can unfreeze all other layers in the model (or keep certain layers still frozen) and retrain. 
 
 # <codecell>
 
-classifier = AmazonKerasClassifier(preprocessor)
-classifier.add_conv_layer()
-classifier.add_flatten_layer()
-classifier.add_ann_layer(len(preprocessor.y_map))
+from keras.optimizers import Adam
+from keras.callbacks import ModelCheckpoint, EarlyStopping, ReduceLROnPlateau
 
-train_losses, val_losses = [], []
-for learn_rate, epochs in zip(learn_rates, epochs_arr):
-    tmp_train_losses, tmp_val_losses, fbeta_score = classifier.train_model(learn_rate, epochs, batch_size, 
-                                                                           train_callbacks=[checkpoint, fbeta, csv])
-    train_losses += tmp_train_losses
-    val_losses += tmp_val_losses
+callbacks = [EarlyStopping(monitor='val_loss', patience=2, verbose=1, min_delta=1e-4),
+             ReduceLROnPlateau(monitor='val_loss', factor=0.5, patience=1, cooldown=0, verbose=1),
```

Comment:
I actually had originally thought it would still be nice to have a "learning rate schedule" on top of the `ReduceLROnPlateau`. Right now I'm retraining without the loop to see how `ReduceLROnPlateau` does on its own, but in this case I changed `factor` to 0.1 so that the learning rate decreases more (effectively creating our learning rate schedule on the fly).